autovalet_system/src/autovalet_system.py

```diff
@@ -0,0 +1,246 @@
+        self.color_sub       = Subscriber(self.color_topic,Image)
+        self.depth_sub       = Subscriber(self.depth_topic,Image)
+        self.ats             = ApproximateTimeSynchronizer([self.color_sub, self.depth_sub], queue_size=2, slop=0.5)
+        self.ego_line        = None
+        self.depth_frame_id  = None
+        self.depth_img       = None
+        self.color_img       = None
+        self.ats.registerCallback(self.registered_image_callback)
+
+        # time tracker for publishing goals at a lower rate
+        self.previous_time = rospy.get_time()
+        
+        self.laneDetector = self.init_detector(self.colorInfo_topic, 
+                                               self.laneCloud_topic,
+                                               self.egoLine_topic)
+        
+        # goal gen setup ##########################################################
+        self.goal_topic         = "/move_base_simple/goal"
+        self.map_frame          = "map"
+        self.goal_pub           = rospy.Publisher(self.goal_topic,PoseStamped,queue_size=1)
+        self.goalGenerator      = goal_generator(self.map_frame)
+        self.current_goal       = PoseStamped()
+
+        # parking setup ###########################################################
+        # tag_topic - name of the topic in which the pose of the marker is being published
+        # husky_frame is the name of the husky's frame we want to keep track of
+        # aruco_frame_name is the NAME OF THE FRAME PUBLISHED AS THE ARUCO TAG
+        self.tag_topic          = '/ARUCO/pose'
+        self.husky_frame        = 'base_link'
+        self.aruco_frame_name   = 'parking_spot' #'aruco_marker_frame' or 'parking_spot'
+                            # (Needs to be same as what is set in aruco launcher)
+        
+        self.parker = parking_spot(self.goal_topic,
+                                   self.tag_topic,
+                                   self.map_frame,
+                                   self.husky_frame,
+                                   self.aruco_frame_name,
+                                   debug=False)
+        
+        # State Machine variables
+        self.current_state = State.START
+        self.prev_state = None
+
+    # helper fxn to load the correct lane detection params and initialize LaneDetector class
+    def init_detector(self, colorInfo_topic, laneCloud_topic, egoLine_topic):
+        yaml_path = rospkg.RosPack().get_path('autovalet_lane_detection')
+        yaml_path += "/config/"
+        if self.sim:
+            yaml_path += "sim_params.yaml"
+        else:
+            yaml_path += "hardware_params.yaml"
+        
+        f = open(yaml_path,'r')
+        yaml_file = yaml.load(f)
+        f.close()
+        rosparam.upload_params('/lane_detection/',yaml_file)
+
+        hlsBounds = rospy.get_param('/lane_detection/hlsbounds')
+        lineParams = rospy.get_param('/lane_detection/lineparams')
+
```

Comment:
so the way I am storing the params actually made the integration more tedious? so many lines to get the params 😭

Reply:
i mean i think yaml files are teh right way to do it. this was kind of a pain to figure out, but now it's done and works as expected. no harm done

Reply:
usually youd load the yaml files into the ros parameter server from a launch file, which is easier. but we wanted to do this in code so it's slightly more involved